src/controllers/dynakube/version/updater.go

```diff
@@ -115,14 +114,10 @@ func setImageIDWithDigest( //nolint:revive
 	if canonRef, ok := ref.(reference.Canonical); ok {
 		target.ImageID = canonRef.String()
 	} else if taggedRef, ok := ref.(reference.NamedTagged); ok {
-		imageVersion, err := imageVersionFunc(ctx, imageUri, dockerCfg)
+		imageVersion, err := imageVersionFunc(ctx, &registry.GoContainerRegistryClient{}, imageUri, dockerCfg)
```

Comment:
```suggestion
		imageVersion, err := imageVersionFunc(ctx, &registry.Client{}, imageUri, dockerCfg)
```
Since the package already has a name registry - it's okay to name it short as Client. and rename interface to not clash with client.